src/types.ts

```
@@ -1,11 +1,21 @@
+export type LogType = (typeof LOG_TYPES)[number];
+export type Severity = (typeof SEVERITIES)[number];
 export interface StackFrame {
   col?: number;
   file?: string;
   line?: number;
   method?: string;
 }
+
+type ClassProperties<C> = {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type -- This is a valid type
+  [Key in keyof C as C[Key] extends Function ? never : Key]: C extends Record<Key, C[Key]>
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)